### PR TITLE
Support Versioning

### PR DESCRIPTION
**Pull request by yosriady**
> Closes https://github.com/comit-network/comit.network/issues/46

This PR adds [versioning](https://v2.docusaurus.io/docs/versioning). Note that it's versioning for both the SDK and handwritten documentation. We can update docs of individual versions manually if need be by modifyng the files in `versioned_docs/`. The versions currently here are :

- `next`: This is content in the `docs/` directory where you'll only write in. Content in this directory is published as a draft that generally is not for public viewing. It's accessible with a `https://comit.network/docs/next` URL prefix on the website.
- `0.15.2` is the latest stable version. The highest numbered version is 'stable'. This is the default published version when you go to `https://comit.network/docs` URL. The content source lives in `versioned_docs/version-0.15.2/`
- `0.14.1` and `0.13.0` can be accessed from the dropdown and have a URL prefix e.g. `https://comit.network/docs/0.14.1`. The content source lives in `versioned_docs/version-0.14.1/` and `versioned_docs/version-0.13.0/` respectively. 

Screenshot 2020-04-08 at 3 07 30 PM

The workflow is explained here: https://github.com/comit-network/comit.network/pull/53#issuecomment-610787371

I also added a Versions page and dropdown component.

> Dropdown

Screenshot 2020-04-08 at 2 00 00 PM

> Versions are displayed on each page

Screenshot 2020-04-08 at 1 57 57 PM

> Versions index page

Screenshot 2020-04-08 at 1 47 14 PM

### CODE REVIEW

**Comment by da-kami** [comment]
So the version is currently linked to the SDK? So the complete documentation - including handwritten documentation - depends on the SDK? 

But not all hand written documentation is versioned, correct?

**Comment by yosriady** [comment]
@da-kami The Docusaurus approach to versioning involves making a 'snapshot' at specific tags. This means all handwritten documentation is versioned. Right now, the handwritten documentation across all versions is duplicated so they are the same. 

From a technical writer perspective, they only have to update once, and only update the latest documentation. When a new release is created, the latest documentation at that point in time is 'archived' and saved.

**Comment by yosriady** [comment]
@da-kami To minimize confusion, we can remove the handwritten documentation for older versions and only keep the SDK documentation for older versions.

**Comment by da-kami** [comment]
> @da-kami The Docusaurus approach to versioning involves making a 'snapshot' at specific tags. This means all handwritten documentation is versioned. Right now, the handwritten documentation across all versions is duplicated so they are the same.
> 
> From a technical writer perspective, they only have to update once, and only update the latest documentation. When a new release is created, the latest documentation at that point in time is 'archived' and saved.

As far as I understood from the [documentation](https://v2.docusaurus.io/docs/versioning/) there can be `docs` and `versioned_docs` and only folders/files in `versioned_docs` would be versioned. Do we really have to version all docs?

Side-note: Some hand-written documentation (like the tutorial) could actually be versioned. 

The version would not be a "documentation" version, but the SDK's version, correct?

**Comment by yosriady** [comment]
Preview: https://deploy-preview-53--comit-network.netlify.com/docs/getting-started/create-comit-app/

**Comment by yosriady** [comment]
> > @da-kami The Docusaurus approach to versioning involves making a 'snapshot' at specific tags. This means all handwritten documentation is versioned. Right now, the handwritten documentation across all versions is duplicated so they are the same.
> > From a technical writer perspective, they only have to update once, and only update the latest documentation. When a new release is created, the latest documentation at that point in time is 'archived' and saved.
> 
> As far as I understood from the [documentation](https://v2.docusaurus.io/docs/versioning/) there can be `docs` and `versioned_docs` and only folders/files in `versioned_docs` would be versioned. Do we really have to version all docs?
> 
> Side-note: Some hand-written documentation (like the tutorial) could actually be versioned.
> 
> The version would not be a "documentation" version, but the SDK's version, correct?

I think there's no harm to versioning docs in this way, since we only need to update one set of documentation at a time.

Yes, the version is not a "documentation" version, but the SDK's version. This lets us tie specific documentation to a particular version.

Docusaurus basically copies the current `docs/` folder to `versioned_docs/` whenever you run `docusaurus version <tag>`.

**Comment by yosriady** [comment]
# Workflow

- You only modify files in `docs/`
- When you want to publish your changes, run `yarn docusaurus docs:version 0.15.3`. 

> Note: to republish the documentation for an old version e.g. `0.15.2` you must first remove the version you want to overwrite from [`versions.json`](https://github.com/comit-network/comit.network/blob/3cc13e309035d5a97751009c26e97e7a3c192ad0/versions.json). If it's a new version, you don't have to do this. The documentation for that version will update based on the current `docs/`.

- The above command will create a `version-0.15.3` folder in `versioned_docs` and `version-0.15.3-sidebars.json` in `versioned_sidebars`. Think of these files as versioned snapshots of the documentation.
- That's it. You can commit and push.

**Comment by yosriady** [comment]
Screenshot 2020-04-08 at 3 07 30 PM

File: generateSidebar.js [hunk 1]
```diff
@@ -1,24 +1,41 @@
 const _ = require('lodash');
 const fs = require('fs');
-const { docs: toc } = require('./sidebars.temp.js')
-const dest = require('./sdkSidebar.json')
 
-const formatCategory = (category) => {
+// Note: A 'sidebar' is a Table of Contents for a specific SDK version.
```
Comment (yosriady) [inline generateSidebar.js:4]: I'm just refactoring a bit here.

File: docusaurus.config.js [hunk 2]
```diff
@@ -23,8 +25,28 @@ module.exports = {
...
       },
       links: [
-        {to: 'docs/getting-started/create-comit-app', label: 'Docs', position: 'left'},
+        {
+          to: 'docs/getting-started/create-comit-app', 
+          label: 'Docs', 
+          position: 'left',
+        },
         {to: 'blog', label: 'Blog', position: 'left'},
+        {
```
Comment (yosriady) [inline docusaurus.config.js:34]: Adds versions dropdown.

File: package.json [hunk 3]
```diff
@@ -10,8 +10,8 @@
     "build:sdk": "./build.sh"
   },
   "dependencies": {
-    "@docusaurus/core": "^2.0.0-alpha.43",
-    "@docusaurus/preset-classic": "^2.0.0-alpha.43",
+    "@docusaurus/core": "^2.0.0-alpha.50",
```
Comment (yosriady) [inline package.json:13]: Upgrades Docusaurus.

File: src/pages/versions.js [hunk 4]
```diff
@@ -0,0 +1,96 @@
+import React from 'react';
```
Comment (yosriady) [inline src/pages/versions.js:1]: New Versions page.

**Comment by yosriady** [comment]
> Looks good!
> 
> Please note most link on the [version page](https://deploy-preview-53--comit-network.netlify.com/versions/) are broken.

Good catch! Links fixed in the latest commit.

File: package.json (hunk 3 above)
Comment (da-kami) [reply package.json:13]: sweet :)

**Pull request merged**